standard/template/generator-template/component/{name_pc}/{name_pc}.tsx.sg

```diff
@@ -0,0 +1,8 @@
+import React from 'react';
+import { Styled{{name_pc}} } from './{{name_pc}}.styles';
+
+type {{name_pc}}Props = {};
+
+export default function {{name_pc}}(props: {{name_pc}}Props) {
+  return <Styled{{name_pc}} {...props}>{{name_pc}}</Styled{{name_pc}}>;
```

Comment:
I guess this is the only way to properly pass all props to the Styled container, but this also means we won't destructure the individual props, leaving us with a lot of `props.x` in the code?

Does forwarding all props risk them showing up on the DOM element? If so should we do something with https://medium.com/@probablyup/introducing-transient-props-f35fd5203e0c?

Reply:
> I guess this is the only way to properly pass all props to the Styled container, but this also means we won't destructure the individual props, leaving us with a lot of props.x in the code?

This is just a shortcut, you can easily destructure props by changing
```
function {{name_pc}}(props: {{name_pc}}Props) { ... }
```
into
```
function {{name_pc}}({prop1, prop2, ...props}: {{name_pc}}Props) { ... }
```
Perhaps we could just add the destructure syntax from the get go:
```
function {{name_pc}}({ ...props}: {{name_pc}}Props) { ... }
```

> Does forwarding all props risk them showing up on the DOM element? If so should we do something with https://medium.com/@probablyup/introducing-transient-props-f35fd5203e0c?

It's a risk, but a risk that the developer has to handle on his own. Either by using transient props, using non clashing names or what we did with OBS - add another component to destructure the "invalid" props out.

Reply:
The "default safe way" would be to manually only pass the relevant props to the Styled Component. I'd rather do that (and destructuring props) by default in our template.

Although I remember @nswaldman doing something similar in our recent project without much issues. And was even needed to pass `className` down when extending those components. So maybe we're fine.

Reply:
@jesse-mm also used transient props on a recent project, which had components spreading all props into the styled component and there was no issues. I think as long as we understand that this is a possibility and have a proper pattern in place to avoid it, then we should be fine.

Reply:
Yes transient props worked perfectly fine for us :), and I like it as it indicates those are style related props.